src/Facebook/FileUpload/FacebookResumableUploader.php

```diff
@@ -0,0 +1,201 @@
+            $filePath,
+            $res['upload_session_id'],
+            $res['video_id'],
+            $res['start_offset'],
+            $res['end_offset'] - $res['start_offset']
+        );
+
+        return $firstTransferChunk;
+    }
+
+    /**
+     * Upload - phase transfer
+     *
+     * @param $endpoint
+     * @param FacebookTransferChunk $chunk
+     * @return FacebookTransferChunk
+     *
+     * @throws FacebookResumableUploadException
+     * @throws FacebookSDKException
+     */
+    public function transfer($endpoint, FacebookTransferChunk $chunk)
+    {
+        $transReqParams = [
+            'upload_phase' => 'transfer',
+            'upload_session_id' => $chunk->getUploadSessionId(),
+            'start_offset' => $chunk->getStartOffset(),
+            'video_file_chunk' => $chunk,
+        ];
+
+        $maxTransferTries = $this->maxTransferTries;
+
+        $request = new FacebookRequest(
+            $this->app,
+            $this->accessToken,
+            'POST',
+            $endpoint,
+            $transReqParams
+        );
+
+        while ($maxTransferTries > 0) {
+            try {
+                $res= $this->client->sendRequest($request)->getDecodedBody();
+
+                $nextTransferChunk = new FacebookTransferChunk(
+                    $chunk->getFilePath(),
+                    $chunk->getUploadSessionId(),
+                    $chunk->getVideoId(),
+                    $res['start_offset'],
+                    $res['end_offset'] - $res['start_offset']
+                );
+
+                return $nextTransferChunk;
+            } catch (FacebookSDKException $e) {
+                $preException = $e->getPrevious();
+                if ($preException instanceof FacebookResumableUploadException) {
+                    if (--$maxTransferTries <= 0) {
+                        $resumeContext = new FacebookResumeContext(
+                            $endpoint,
+                            $this->accessToken,
+                            $chunk
+                        );
+                        $preException->setResumeContext($resumeContext);
+                        throw $e;
+                    }
+                } else {
+                    throw $e;
+                }
+                continue;
+            }
+        }
+    }
+
+    /**
+     * Upload - phase finish
+     *
+     * @param $endpoint
+     * @param $uploadSessionId
+     * @return boolean
+     *
+     * @throws FacebookSDKException
+     */
+    public function finish($endpoint, $uploadSessionId)
+    {
+        $finishReqParams = [
+            'upload_phase' => 'finish',
+            'upload_session_id' => $uploadSessionId,
+        ];
+
+        $request = new FacebookRequest(
+            $this->app,
+            $this->accessToken,
+            'POST',
+            $endpoint,
+            $finishReqParams
+        );
+
+        $res = $this->client->sendRequest($request)->getDecodedBody();
```

Comment:
@YuanhuiChen rename all your `$res` to `$response`, `$decodedBody`, `$data`, or any thing else but please, a full word. :)